beacon-chain/blockchain/service.go

```diff
@@ -146,7 +146,23 @@ func (s *Service) Start() {
 	}
 
 	if beaconState == nil {
-		beaconState, err = s.stateGen.StateByRoot(s.ctx, bytesutil.ToBytes32(cp.Root))
+		r := bytesutil.ToBytes32(cp.Root)
+		// Before the first finalized epoch, in the current epoch,
+		// the finalized root is defined as zero hashes instead of genesis root hash.
+		// We want to use genesis root to retrieve for state.
+		if r == params.BeaconConfig().ZeroHash {
+			genesisBlock, err := s.beaconDB.GenesisBlock(s.ctx)
+			if err != nil {
+				log.Fatalf("Could not fetch finalized cp: %v", err)
+			}
+			if genesisBlock != nil {
```

Comment:
if genesis block is nil we will still end up doing `StateByRoot` , which I am assuming isnt the desired behaviour.

Reply:
That's fine. We want the beacon state to be `nil` then so we can enter `Waiting to reach the validator deposit threshold to start the beacon chain...` code block.

I've tested this case as well